lib/proformaxml/services/importer.rb

```diff
@@ -46,11 +46,16 @@ def filestring_from_zip(filename)
       end
     end
 
+    def remove_referenced_files
+      @task.files.reject! {|f| (@task.tests.map(&:files) + @task.model_solutions.map(&:files)).flatten.include? f }
```

Comment:
This one liner doesn't seem to be very performant, is it? I would assume we are performing the `@task.tests.map(&:files) + @task.model_solutions.map(&:files)).flatten` part for each `@task.files`, which seems unnecessary. Do you think we can improve that?